### ISSUE TITLE
Multiplying with `*` symbol only works in one direction

### ISSUE THREAD

**Issue by etiennebacher** [pola-rs/r-polars#344]
I can't use  `*` with e.g `pl$max()`:

``` r
library(polars)
packageVersion("polars")
#> [1] '0.7.0.9000'

df = pl$DataFrame(x = c(0, 0, 1), y = c(1, 2, 3))

# fine
df$filter(pl$col("y") > pl$max("y")$mul(0.5))
#> shape: (2, 2)
#> ┌─────┬─────┐
#> │ x   ┆ y   │
#> │ --- ┆ --- │
#> │ f64 ┆ f64 │
#> ╞═════╪═════╡
#> │ 0.0 ┆ 2.0 │
#> │ 1.0 ┆ 3.0 │
#> └─────┴─────┘

# error
df$filter(pl$col("y") > 0.5 * pl$max("y"))
#> Error in e1$mul: $ operator is invalid for atomic vectors
```

But it works in `py-polars`:
```python
>>> import polars as pl
>>> df = pl.DataFrame(dict(
...     x=[0, 0, 1, 1], 
...     y=[1, 2, 3, 3]) 
... )
>>>
>>> df.filter((pl.col("y") > 0.5 * pl.max("y")))
shape: (3, 2)
┌─────┬─────┐
│ x   ┆ y   │
│ --- ┆ --- │
│ i64 ┆ i64 │
╞═════╪═════╡
│ 0   ┆ 2   │
│ 1   ┆ 3   │
│ 1   ┆ 3   │
└─────┴─────┘
```

**Comment by etiennebacher** [comment]
Due to `*.Expr` only working when the first term is an `Expr`. Simplified reprex:

``` r
library(polars)

pl$max("y") * 0.5
#> polars Expr: [(col("y").max()) * (0.5)]

0.5 * pl$max("y")
#> Error in e1$mul: $ operator is invalid for atomic vectors
```

**Comment by etiennebacher** [comment]
Closed with #346
